File: SDKs/shutterstock.com/1.1.5/python/src/sdk/models/operations/deleteimagecollectionitems.py
import dataclasses
from typing import Optional
from ..shared import security as shared_security


@dataclasses.dataclass
class DeleteImageCollectionItemsPathParams:
    id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'id', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class DeleteImageCollectionItemsQueryParams:
    item_id: Optional[list[str]] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'item_id', 'style': 'form', 'explode': True }})
    

@dataclasses.dataclass
class DeleteImageCollectionItemsSecurity:
    customer_access_code: shared_security.SchemeCustomerAccessCode = dataclasses.field(metadata={'security': { 'scheme': True, 'type': 'oauth2' }})
    

@dataclasses.dataclass
class DeleteImageCollectionItemsRequest:
    path_params: DeleteImageCollectionItemsPathParams = dataclasses.field()
    query_params: DeleteImageCollectionItemsQueryParams = dataclasses.field()
    security: DeleteImageCollectionItemsSecurity = dataclasses.field()
    

@dataclasses.dataclass
class DeleteImageCollectionItemsResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    

File: SDKs/shutterstock.com/1.1.5/typescript/src/sdk/models/operations/deleteimagecollectionitems.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class DeleteImageCollectionItemsPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=id" })
  id: string;
}


export class DeleteImageCollectionItemsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=item_id" })
  itemId?: string[];
}


export class DeleteImageCollectionItemsSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  customerAccessCode: shared.SchemeCustomerAccessCode;
}


export class DeleteImageCollectionItemsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: DeleteImageCollectionItemsPathParams;

  @SpeakeasyMetadata()
  queryParams: DeleteImageCollectionItemsQueryParams;

  @SpeakeasyMetadata()
  security: DeleteImageCollectionItemsSecurity;
}


export class DeleteImageCollectionItemsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
